lte/gateway/docker/mme/Dockerfile.ubuntu18.04

```diff
@@ -116,94 +123,131 @@ RUN git clone --recurse-submodules -b v1.15.0 https://github.com/grpc/grpc && \
+    make install && \
+    make clean 2>&1 > /dev/null && \
+    cd / && \
+    rm -Rf gnutls-3.1.23.tar.xz && \
+    ldconfig -v
 
 ##### liblfds
 # https://www.liblfds.org/mediawiki/index.php?title=r7.1.0:Building_Guide_(liblfds)
-RUN git clone https://github.com/liblfds/liblfds.git && \
-    cd liblfds/liblfds/liblfds7.1.0/liblfds710/build/gcc_gnumake/ && \
+RUN cd /liblfds/liblfds/liblfds7.1.0/liblfds710/build/gcc_gnumake/ && \
     make -j`nproc` && \
-    make ar_install
+    make ar_install && \
+    make clean
 
 ##### libgtpnl
 # review https://github.com/OPENAIRINTERFACE/openair-cn/blob/master/build/tools/build_helper.gtpnl
-RUN git clone https://git.osmocom.org/libgtpnl && \
-    cd libgtpnl && \
+RUN cd libgtpnl && \
+    # Moved git clone https://git.osmocom.org/libgtpnl && \
     git reset --hard 345d687 && \
     autoreconf -fi && \
     ./configure && \
     make -j`nproc` && \
     make install && \
+    make clean 2>&1 > /dev/null && \
     ldconfig
 
 #####  asn1c
-RUN git clone https://gitlab.eurecom.fr/oai/asn1c.git && \
-    cd asn1c && \
+RUN cd asn1c && \
+    # Moved git clone https://github.com/OPENAIRINTERFACE/asn1c.git && \
     git checkout f12568d617dbf48497588f8e227d70388fa217c9 && \
     autoreconf -iv && \
     ./configure && \
     make -j`nproc` && \
     make install && \
-    ldconfig
-
-
-# Add Converged MME sources to the container
-COPY ./ $MAGMA_ROOT
+    git clean -x -d -ff -q
 
 ##### FreeDiameter
-RUN git clone https://github.com/OPENAIRINTERFACE/opencord.org.freeDiameter.git freediameter && \
-    cd freediameter && \
-    patch -p1 < $MAGMA_ROOT/lte/gateway/c/core/oai/patches/0001-opencoord.org.freeDiameter.patch && \
-    patch -p1 < $MAGMA_ROOT/lte/gateway/c/core/oai/patches/0002-opencoord.org.freeDiameter.patch && \
-    mkdir build && \
-    cd build && \
-    cmake ../ && \
+RUN cd freediameter && \
+    # Moved git clone https://github.com/OPENAIRINTERFACE/opencord.org.freeDiameter.git freediameter && \
+    git pull origin master && \
+    git log -n1 && \
+    echo "Patching dict_S6as6d" && \
+    patch -p1 < /patches/0001-opencoord.org.freeDiameter.patch && \
+    patch -p1 < /patches/0002-opencoord.org.freeDiameter.patch && \
+    mkdir _build && \
+    cd _build && \
+    cmake -DBUILD_TESTING=false ../ && \
+    grep DISABLE_SCTP CMakeCache.txt && \
     awk '{if (/^DISABLE_SCTP/) gsub(/OFF/, "ON"); print}' CMakeCache.txt > tmp && mv tmp CMakeCache.txt && \
+    grep DISABLE_SCTP CMakeCache.txt && \
     make -j`nproc` && \
-    make install
+    make install && \
+    cd ../ && rm -Rf _build && \
+    ldconfig --verbose
 
 # Install bazel
 WORKDIR /usr/sbin
 RUN wget --progress=dot:giga https://github.com/bazelbuild/bazelisk/releases/download/v1.10.0/bazelisk-linux-amd64 && \
     chmod +x bazelisk-linux-amd64 && \
     ln -sf /usr/sbin/bazelisk-linux-amd64 /usr/sbin/bazel
 
+# Starting from here, we are building MAGMA services (sctpd and mme)
+FROM magma-mme-base as magma-mme-builder
```

Comment:
Since we actually have a stage here now, should the `bazel build` not happen once before this in the `magma-mme-base` and then again in the `magma-mme-builder`?